website/docs/guides/testing.md

```diff
@@ -3,70 +3,486 @@ id: testing
 title: Testing
 ---
 
-You can test Effection code with any test framework. Since Effection integrates
-seamlessly with `async/await` code, any test framework which supports
-asynchronous tests works well with Effection.
+Effection not only simplifies the process of writing tests, but by
+managing all the timing complexity of your setup and teardown, it can
+often make new kinds of tests possible that beforehand may have seemed
+unachievable.
 
-Effection itself is tested with [mocha][], and we have created a small
-integration to make the process of writing tests with mocha easier.
+Currently, both [Jest][jest] and [Mocha][mocha] are supported out of the box, but
+because of the way Effection works, and because of the nature of task
+you have to do while testing, it's easy to integrate it with
+any test framework you like.
 
-## Getting started
+## Lifecycle
 
-Add the `@effection/mocha` package:
+The reason Effection can so effectively power your tests is because the
+life-cycle of a test mirrors that of an Effection Task almost perfectly.
 
+Consider that almost every test framework executes the following sequence of
+operations when running a test:
+
+1. setup()
+2. run()
+3. teardown()
+
+While there are certainly different ways of expressing this syntactically, it
+turns out that as a fundamental testing pattern, it is nearly universal. Notice
+that, like tests, the cleanup of an effection operation is intrinsic. We can
+leverage this fact to associate an effection task, or scope, with each test.
```

Comment:
You might want to link to a doc about the nomenclature (if there is one).  The reader might not know what a task, scope or resource is.

Reply:
@dagda1 I made changes to keep all of the discussion referencing the concrete concept of tasks, and halting tasks instead of "scopes" or "disposing of scope"